app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,67 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+# before_action :configure_sign_up_params, only: [:create]
+# before_action :configure_account_update_params, only: [:update]
+
+  # GET /resource/sign_up
+  def new
+    if session[:token].nil? or Invite.find_by(token: session[:token], accepted: false).blank?
+     redirect_to root_path
+    else
+     super
+    end
+  end
+
+  # POST /resource
+  def create
+     super
+     if resource.persisted?  #if user has registered
+       Invite.find_by(token: session[:token]).update_attributes(accepted: true)
+     end
+  end
+
+  # GET /resource/edit
+  # def edit
+  #   super
+  # end
+
+  # PUT /resource
+  # def update
+  #   super
+  # end
+
+  # DELETE /resource
+  # def destroy
+  #   super
+  # end
+
+  # GET /resource/cancel
+  # Forces the session data which is usually expired after sign
+  # in to be expired now. This is useful if the user wants to
+  # cancel oauth signing in/up in the middle of the process,
+  # removing all OAuth session data.
+  # def cancel
+  #   super
+  # end
+
+  # protected
+
+  # If you have extra params to permit, append them to the sanitizer.
+  # def configure_sign_up_params
+  #   devise_parameter_sanitizer.permit(:sign_up, keys: [:attribute])
+  # end
+
+  # If you have extra params to permit, append them to the sanitizer.
+  # def configure_account_update_params
+  #   devise_parameter_sanitizer.permit(:account_update, keys: [:attribute])
+  # end
+
+  # The path used after sign up.
+  # def after_sign_up_path_for(resource)
+  #   super(resource)
+  # end
+
+  # The path used after sign up for inactive accounts.
+  # def after_inactive_sign_up_path_for(resource)
+  #   super(resource)
+  # end
```

Comment:
Delete the autogenerated commented code. We can look these up in the gem or the docs if needed later.